core/src/main/scala/kafka/log/Log.scala

```diff
@@ -436,8 +466,8 @@ class Log(@volatile var dir: File,
     * @throws KafkaStorageException If the append fails due to an I/O error.
     * @return Information about the appended messages including the first and last offset.
     */
-  def appendAsLeader(records: MemoryRecords, leaderEpoch: Int): LogAppendInfo = {
-    append(records, assignOffsets = true, leaderEpoch)
+  def appendAsLeader(records: MemoryRecords, leaderEpoch: Int, isFromClient: Boolean = true): LogAppendInfo = {
```

Comment:
A general comment about default values: we should consider carefully when to use them as they are a common source of bugs. When we use them, the compiler no longer ensures that we think about what the value should be.